go/vt/vtgate/executorcontext/safe_session.go

```diff
@@ -410,15 +468,22 @@ func (session *SafeSession) findSessionLocked(keyspace, shard string, tabletType
 	return nil
 }
 
+type myShardActionInfo interface {
```

Comment:
myShardActionInfo doesn't sound like a great name. Why the `my`. Why not just `shardActionInfo`